WPMedia/fusion-news-theme-blocks
# Tmedia 595 - Results List with Collection cannot have more than 10 entries

## Description
When using the Results List Block from Themes with the Collections content-source, you cannot get a size over 10, it will just show nothing instead.
Considering the collections-api can serve up to 20 entries, I don't understand why.
I was able to reproduce it on 2 environments, Sandbox. All OOTB Themes, none of the examples are currently live - but it can be easily reproduced on Themes environment.

## Jira Ticket
[- [TMEDIA-](https://arcpublishing.atlassian.net/browse/TMEDIA-)](https://arcpublishing.atlassian.net/browse/TMEDIA-595)

## Acceptance Criteria
Any size up to 20 displays in the Results List with a Collection.

## Test Steps
- Add test steps a reviewer must complete to test this PR

1. Checkout this branch `git checkout tmedia-595`
2. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/collections-content-source-block,@wpmedia/resizer-image-block,@wpmedia/results-list-block
`
3. Create a new page or use and existing one and ensure a Results List Block is added.
4. In the configure content section for the result list, add the following properties:
![image](https://user-images.githubusercontent.com/833461/147894257-172c6da9-4328-4a0e-8a4a-37de7d88b3c9.png)
5. Save and publish changes and then view the published page under the Gazette profile.


## Effect Of Changes
### Before
Any setting over 10 would not show any entries in the result list.


### After
Any setting over 10 is now shown.  However, if the entry is over 20, the number is trimmed to 20 because of the limitation of the collection API being capped at 20 entries.

## Dependencies or Side Effects
If the user configures the results list with 20 entries, the show more button is not shown.  I was unable to fix this. One thing I tried is that since an entry of 10-19 was showing the show more button, I tried to cap an entry of 20 to 19, but that did not seem to work (probably due to client hydration taking over?).  

## Author Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
- [x ] Confirmed all the test steps a reviewer will follow above are working.
- [x ] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [ x] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [x ] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [x ] Confirmed this PR has unit test files
  - [x ] Ran `npm run test`, made sure all tests are passing


## Reviewer Checklist
_The reviewer of the PR should copy-paste this template into the review comments on review._
